qrexec/tests/socket/daemon.py

```diff
@@ -49,9 +50,27 @@ class TestDaemon(unittest.TestCase):
     POLICY_PROGRAM = """\
 #!/bin/sh
 
+# -- remote_domain_name target_name service_name
 echo "$@" > {tempdir}/qrexec-policy-params
+
 sleep $(cat {tempdir}/qrexec-policy-sleep || echo 0)
-exit $(cat {tempdir}/qrexec-policy-exitcode || echo 1)
+exit_code=$(cat {tempdir}/qrexec-policy-exitcode || echo 1)
+
+# Prepare the response based on the exit code
+if [ "$exit_code" -eq 0 ]; then
+    # Allow response
+    printf 'result=allow\n'
+    printf 'autostart=True\n'
+    printf 'user=toto\n'
+    printf 'target=%s\n' "$2"
```

Comment:
$2 is the source, not target in the 5-arg version (that qrexec-daemon will call)

Reply:
Yes I forgot to ask you, and you answered in the comment below about using self.domain_name instead.